Lib/test/_test_multiprocessing.py

```diff
@@ -2611,6 +2620,41 @@ def errback(exc):
         p.close()
         p.join()
 
+    def test_broken_process_pool1(self):
+        from multiprocessing.pool import BrokenProcessPool
+        p = multiprocessing.Pool(2)
+        res = p.map_async(waiting, range(10))
+        # Kill one of the pool workers.
+        waiting(None)
+        pid = p._pool[0].pid
+        os.kill(pid, signal.SIGTERM)
```

Comment:
Can't you write `p._pool[0].terminate()` instead?

Reply:
Tests are meant to mimic the situation when the linux kernel kills one worker, so the signal is external. Nonetheless, I've added one more test with internal cause (i.e. `p._pool[0].terminate()`).